### PR TITLE
Document Terraform provider

### PR DESCRIPTION
**Pull request by hbokh**
This is a "replacement" for #2283.
Will solve #2166 
Written with Cobbler 3.2.x and Terraform v0.14.x in mind.

### CODE REVIEW

**Comment by hbokh** [comment]
Preview: https://cobbler--2528.org.readthedocs.build/en/2528/user-guide/terraform-provider.html

**Review by SchoolGuy — APPROVED** [review]
LGTM in my eyes. Thanks @hbokh !

File: docs/user-guide/terraform-provider.rst [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+
+.. code::
+
+   resource "cobbler_system" "webserver" {
+     count            = "1"
+     name             = "webserver"
+     profile          = "debian10-x86_64"
+     hostname         = "webserver.example.com"       # Use FQDN
+     autoinstall      = "debian10_VMware.seed"
+     # NOTE: Extra spaces at the end is there for a reason!
```
Comment (dansou901) [inline docs/user-guide/terraform-provider.rst:273]: Extra spaces at the end are there for a reason

File: docs/user-guide/terraform-provider.rst [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+    name_servers        = ["1.1.1.1", "8.8.8.8"]   # Should be a list
+    name_servers_search = ["example.com"]
+    repos               = ["debian10-x86_64"]
+  }
+
+Example configuration - combined
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is also possible to combine multiple resources into one file.
+For example, this will combine a Ubuntu Bionic distro, a profile and a system:
```
Comment (dansou901) [inline docs/user-guide/terraform-provider.rst:370]: this will combine an Ubuntu Bionic distro

**Review by dansou901 — CHANGES_REQUESTED** [review]
See inline comments

**Review by dansou901 — APPROVED** [review]
LGTM now.

**Pull request merged**